Entities/Industry/Drill/Drill.as

```diff
@@ -388,9 +391,44 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
 void onAttach(CBlob@ this, CBlob@ attached, AttachmentPoint @attachedPoint)
 {
 	this.getCurrentScript().runFlags &= ~Script::tick_not_sleeping;
+	this.set_s32("showHeatTo", getPlayerIndex(attached.getPlayer()));
+}
+
+void onDetach(CBlob@ this, CBlob@ detached, AttachmentPoint @attachedPoint)
+{
+	this.set_s32("showHeatTo", -1);
 }
 
 void onThisAddToInventory(CBlob@ this, CBlob@ blob)
 {
 	this.getSprite().SetEmitSoundPaused(true);
 }
+
+void onRender(CSprite@ this)
+{
+	CBlob@ blob = this.getBlob();
+	CPlayer@ holder = getPlayer(blob.get_s32("showHeatTo"));
+	
+	Vec2f mousePos = getControls().getMouseWorldPos();
+	Vec2f blobPos = blob.getPosition();
+
+	if (getLocalPlayer() is null || getLocalPlayer().getBlob() is null)
+		return;
+
+	bool inRange = (blobPos - getLocalPlayer().getBlob().getPosition()).getLength() < max_heatbar_view_range;
+	bool hover = (mousePos - blobPos).getLength() < blob.getRadius() * 1.50f;
+	
+	if ((hover && inRange) || (holder !is null && holder.isLocal()))
+	{
+		u8 heat = blob.get_u8(heat_prop);
+
+		Vec2f pos = blob.getScreenPos() + Vec2f(-21, 10);
+		Vec2f dimension = Vec2f(36, 8);
+
+		f32 percentage = f32(heat) / heat_max;
+
+		GUI::DrawRectangle(Vec2f(pos.x, pos.y), Vec2f(pos.x + dimension.x, pos.y + dimension.y));
+		GUI::DrawRectangle(Vec2f(pos.x + 2, pos.y + 2), Vec2f(pos.x + dimension.x - 2, pos.y + dimension.y - 2), SColor(0xff0ddb1e));
+		GUI::DrawRectangle(Vec2f(pos.x + 2 + (dimension.x-4) * (1 - percentage), pos.y + 2), Vec2f(pos.x + dimension.x - 2, pos.y + dimension.y - 2), SColor(0xffdb0d17));
```

Comment:
Any reason why you couldn't use `GUI::DrawProgressBar(Vec2f upperleft, Vec2f lowerright, float percent)`?

Reply:
I wasn't able to use the ```GUI::DrawProgressBar(Vec2f upperleft, Vec2f lowerright, float percent)``` function because it does not support using custom progressbar colours, I implemented it this way because otherwise I wouldn't be able to make the progress show up in red.

Reply:
My only issue with making a custom progress bar is that would look inconsistent with every other progress bar in the game

Reply:
I agree that it can potentially look odd, though a heat indicator would in my opinion look best if it were red.